src/containers/flows/CancelBid/CancelBid.tsx

```diff
@@ -41,13 +41,13 @@ export const CancelBid = ({
 
 	const onFinish = () => {
 		onClose();
+		onSuccess();
```

Comment:
If `onSuccess` is here, it won't run if the user closes the modal, or if the user waits until the `tx` finishes and closes the modal by clicking out of it, rather than hitting the "Finish" button. Not sure if this is the intended behaviour, but worth considering.

Reply:
Ah yeah that makes sense. I changed it to this to prevent the modal closing once ```tx?.wait()``` was complete in ```useCancelBid```. I've reverted it now but unsure how to prevent the modal closing and therefore skipping the Success step of the flow. Would you have any ideas why the modal is closing as below?

https://user-images.githubusercontent.com/39112648/178272796-d0105b1d-8221-46f1-b3ad-f51a10778fef.mov

Reply:
Ahhh I see - the `onSuccess` will be triggering an API call to get your bids, and since you just cancelled your bid, the button won't render and therefore the modal will disappear.

Could use an `isMounted` ref, and after `tx.wait()` check if `isMounted.current === false`. If yes, call `onSuccess()`, otherwise show success screen.